### PR TITLE
Add box that installs ODL via Puppet deb method on Ubuntu 16.04

### PR DESCRIPTION
**Pull request by AkshitaJha**
Signed-off-by: Akshita Jha <zenith158@gmail.com>

### CODE REVIEW

**Comment by dfarrell07** [comment]
After I rsync your latest code from https://github.com/dfarrell07/puppet-opendaylight/pull/131 to modules/opendaylight, I see `vagrant up ubuntu16_pup_deb` fail with:

    Error: Could not find class apt for ubuntu-xenial.localdomain on node ubuntu-xenial.localdomain

**Comment by dfarrell07** [comment]
This seems to work. Need to get puppet-odl changes finished and merged before this, still manually rsyncing them into place atm.

**Comment by AkshitaJha** [comment]
Removed `ppa:openjdk-r/ppa` from Vagrantfile. Adding that via puppet manifests directly.

File: Vagrantfile [hunk 1]
```diff
@@ -274,4 +274,23 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
   end
 
+  # Box that installs ODL via Puppet deb method on Ubuntu 16.04
+  config.vm.define "ubuntu16_pup_deb" do |ubuntu16_pup_deb|
+    # Build Vagrant box based on Ubuntu 16.04
+    ubuntu16_pup_deb.vm.box = "ubuntu/xenial64"
+
+    # Install Puppet
+    ubuntu16_pup_deb.vm.provision "shell", inline: "apt-get install -y puppet"
+    ubuntu16_pup_deb.vm.provision "shell", inline: "apt-get install -y librarian-puppet"
```
Comment (dfarrell07) [inline Vagrantfile:284]: Might save a bit of time to do these installs in one line

**Comment by dfarrell07** [comment]
Maybe you can squash these commits into one, @AkshitaJha? They do one logical thing, and I don't think we want the intermediary steps to be valid states of the codebase.

**Comment by AkshitaJha** [comment]
> Maybe you can squash these commits into one, @AkshitaJha? They do one logical thing, and I don't think we want the intermediary steps to be valid states of the codebase.

Done.

**Comment by dfarrell07** [comment]
This works - box builds, can SSH in, ODL is running under systemd, can connect to Karaf shell, feature exist. Great job, thanks @AkshitaJha!

**Pull request merged**